integration-tests/oidc-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakTestResource.java

```diff
@@ -71,8 +75,37 @@
         // Invalid
         defineInvalidIntrospectionMockTokenStubForUserWithRoles("expired", emptySet());
 
+        // Login Page
+        server.stubFor(
+                get(urlPathMatching("/auth/realms/quarkus"))
+                        .withQueryParam("redirect_uri", equalTo("http://localhost:8081/code-flow"))
+                        .willReturn(aResponse()
+                                .withHeader("Content-Type", MediaType.TEXT_HTML)
+                                .withBody("<html>\n" +
+                                        "<body>\n" +
+                                        " <form action=\"/login\" name=\"form\">\n" +
+                                        "  <input type=\"text\" id=\"username\" name=\"username\"/>\n" +
+                                        "  <input type=\"password\" id=\"password\" name=\"password\"/>\n" +
+                                        "  <input type=\"hidden\" id=\"state\" name=\"state\" value=\"SOME_STATE\"/>\n" +
```

Comment:
@cemnura yeah, this is good - except that the `state` needs to be captured, for the exact value be eventually returned, so I guess it makes sense to capture the redirect URI as well - as we'd like different code flow tests work against it...

Reply:
I was able to capture and dynamically add the return the state & redirect_uri according to the request.

Apparently,  this can be done on wiremock via [response templating](http://wiremock.org/docs/response-templating/)

Reply:
@cemnura great, we are learning some cool things about Wiremock here :-)